tests/tracer/runtime/test_runtime_metrics.py

```diff
@@ -107,3 +123,63 @@ def test_tracer_metrics(self):
             self.assertRegexpMatches(gauge, "lang_version:")
             self.assertRegexpMatches(gauge, "lang:python")
             self.assertRegexpMatches(gauge, "tracer_version:")
+
+    def test_only_root_span_runtime_internal_span_types(self):
+        with runtime_metrics_service(tracer=self.tracer):
+            for span_type in ("custom", "template", "web", "worker"):
+                with self.start_span("root", span_type=span_type) as root:
+                    with self.start_span("child", child_of=root) as child:
+                        pass
+                assert root.get_tag("language") == "python"
+                assert child.get_tag("language") is None
+
+    def test_only_root_span_runtime_internal_span_types_with_tracer_config(self):
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            self.tracer.configure(collect_metrics=True)
+            for span_type in ("custom", "template", "web", "worker"):
+                with self.start_span("root", span_type=span_type) as root:
+                    with self.start_span("child", child_of=root) as child:
+                        pass
+                assert root.get_tag("language") == "python"
+                assert child.get_tag("language") is None
+            self.tracer.configure(collect_metrics=False)
+
+            self.assertEqual(len(w), 2)
+            self.assertTrue(all(issubclass(_.category, DeprecationWarning) for _ in w))
+
+    def test_span_no_runtime_tags(self):
+        with self.start_span("root") as root:
+            with self.start_span("child", child_of=root.context) as child:
+                pass
+
+        assert root.get_tag("language") is None
+        assert child.get_tag("language") is None
+
+    def test_only_root_span_runtime_external_span_types(self):
+        with runtime_metrics_service(tracer=self.tracer):
+            for span_type in (
+                "algoliasearch.search",
+                "boto",
+                "cache",
+                "cassandra",
+                "elasticsearch",
+                "grpc",
+                "kombu",
+                "http",
+                "memcached",
+                "redis",
+                "sql",
+                "vertica",
+            ):
+                with self.start_span("root", span_type=span_type) as root:
+                    with self.start_span("child", child_of=root) as child:
+                        pass
+                assert root.get_tag("language") is None
+                assert child.get_tag("language") is None
+
+
+def test_fork():
+    if sys.version_info >= (3, 7, 0):
```

Comment:
Why this version requirement?

Reply:
Ah I forgot to use the `forksafe` hook! 🙁